packages/ui-components/README.md

```diff
@@ -43,3 +43,47 @@ The icons are organized into various categories in `./style/icons`, based on whe
   - launcher
   - statusbar
   - settingeditor
+
+## Notes on design of JLIcon
+
+Design goals for JLIcon (already implemented):
+
+- the one true icon
+  - create a single, canonical, simple (as possible) way to set up and use icons in jlab
+- every icon is a symbol
+  - each icon is defined exactly once, and is thereafter referred to via a unique symbol
+  - to use an icon `fooIcon` outside of the file in which it is defined, the pattern will be to import `fooIcon`
+  - this enables compile-time checking; helps to ensure that icons are specified correctly when used (as opposed to the old pattern of specifying icon via a `string` with name or className)
+- every icon is flexible
+  - can used in any context in which icons are used in jlab
+    - every icon can be passsed into Lumino
+    - every icon can create a DOM node
+    - every icon can create a React component
+- dynamic lookup (for when absolutely needed)
+  - Use dynamic lookup for the few cases in which an icon can only be specified as a string (such as in json schema files)
```

Comment:
So the string ids need to be unique? So we probably want to encourage namespacing the strings with the package name, like `'ui-components:myicon'`?